pkg/utils/k8sutil.go

```diff
@@ -475,6 +475,7 @@ func CreateIngress(client kubernetes.Interface, httpTriggerObj *kubelessApi.HTTP
 			Namespace:       httpTriggerObj.Namespace,
 			OwnerReferences: or,
 			Labels:          httpTriggerObj.ObjectMeta.Labels,
+			Annotations:     httpTriggerObj.ObjectMeta.Annotations,
```

Comment:
We may have case (in future) annotation on `HTTPTrigger` may have different purpose. So does it make sense to have expected annotations more explicit as part of `HTTPTrigger` specification as in as a field of the spec, rather than annotations on the `HTTPTrigger` object?

Reply:
Something like below?
````
apiVersion: kubeless.io/v1beta1
kind: HTTPTrigger
metadata:
  name: my-trigger
  namespace: test
spec:
  function: my-function
  host: "test.ingress.host"
  path: "/my-function"
  route: "my-trigger"
  tls: true
  auth:
    metadata:
      annotations:
        "kubeless.io/ingress.class:[traefik|nginx|voyager]"
    spec:
      type: "basic"
      realm: "Authentication required ..."
      secret: "secret-basic-auth"
````

Reply:
some thing like:

````
apiVersion: kubeless.io/v1beta1
kind: HTTPTrigger
metadata:
  name: my-trigger
  namespace: test
spec:
  function: my-function
  host: "test.ingress.host"
  path: "/my-function"
  route: "my-trigger"
  tls: true
  ingress-annotations:
      "kubeless.io/ingress.class" :"traefik"
      "ingress.kubernetes.io/rewrite-target: /
````
httptrigger.spec.ingress-annotations which can be copied directly to kubernetes ingress resource object. It's not elegent in any way. Other options is to let Ingress spec as part of http trigger spec (like we do with deployment, service spec in function spec) then it makes all the fields (path, host, tls etc) redundent.

@andresmgot any preference?

Reply:
It seems that we are treating the `HTTPTrigger` object like something generic but with this step we are tying it back to an Ingress object. 

Users can always create their own ingress objects so bypassing their properties is not that useful. In my opinion if we want to provide basic auth we can have a single parameter: `basic-auth-secret` and we translate it to ingress annotations. That way we provide an API that users can use regardless of the mechanism Kubeless use to provide authentication.

Reply:
> we can have a single parameter: basic-auth-secret and we translate it to ingress 

So the problem is configuration varies from Nginx, Traefik to voyager etc so there is no one stright forward translation. If we dont want to by-pass Kubeless, then if Kubeless can get the Ingress controller it is dealing with it can configure as per the ingress controller. Or let the user pass the customization like in this PR, Kubeless just copies those annotation to the ingress.

Reply:
I would add then a second parameter to know the provider (nginx, traefik or voyager) and specify which ones we support. This trigger could be in theory extended to support other providers (not necessarily based on ingress object) and parameters related to ingress would not apply.

Reply:
So the summarized latest proposal would be something like below:
````
apiVersion: kubeless.io/v1beta1
kind: HTTPTrigger
metadata:
  name: my-trigger
  namespace: test
spec:
  function: my-function
  host: "test.ingress.host"
  path: "/my-function"
  route: "my-trigger"
  tls: true
  basic-auth-secret: "basic-auth-secret"
  basic-auth-type: "traefik"
````